linkerd/app/core/src/config.rs

```diff
@@ -1,1137 +1,115 @@
-
-    // TCP Keepalive set on accepted outbound connections.
-    pub outbound_accept_keepalive: Option<Duration>,
-
-    // TCP Keepalive set on inbound connections to the local application.
-    pub inbound_connect_keepalive: Option<Duration>,
-
-    // TCP Keepalive set on outbound connections to the remote peers.
-    pub outbound_connect_keepalive: Option<Duration>,
-
-    pub inbound_ports_disable_protocol_detection: IndexSet<u16>,
-
-    pub outbound_ports_disable_protocol_detection: IndexSet<u16>,
-
-    pub inbound_router_capacity: usize,
-
-    pub outbound_router_capacity: usize,
-
-    pub inbound_router_max_idle_age: Duration,
-
-    pub outbound_router_max_idle_age: Duration,
-
-    pub inbound_max_requests_in_flight: usize,
-
-    pub outbound_max_requests_in_flight: usize,
-
-    /// Age after which metrics may be dropped.
-    pub metrics_retain_idle: Duration,
-
-    /// Settings for the back-off used to determine the amount of time to wait
-    /// between when encountering errors talking to control plane before
-    /// a new connection is attempted.
-    pub control_backoff: ExponentialBackoff,
-
-    /// The maximum amount of time to wait for a connection to the controller.
-    pub control_connect_timeout: Duration,
-
-    pub identity_config: tls::Conditional<identity::Config>,
-
-    //
-    // Destination Config
-    //
-    /// Where to talk to the control plane.
-    pub destination_addr: ControlAddr,
-
-    pub hostname: Option<String>,
-
-    pub trace_collector_addr: Option<ControlAddr>,
-
-    /// The maximum number of queries to the Destination service which may be
-    /// active concurrently.
-    pub destination_buffer_capacity: usize,
-
-    /// Configured by `ENV_DESTINATION_GET_SUFFIXES`.
-    pub destination_get_suffixes: Vec<dns::Suffix>,
-
-    /// Configured by `ENV_DESTINATION_PROFILE_SUFFIXES`.
-    pub destination_profile_suffixes: Vec<dns::Suffix>,
-
-    /// This token is passed to the Destination service so that it can return
-    /// different results depending on the identity of the proxy making the
-    /// call.
-    pub destination_context: String,
-
-    //
-    // DNS Config
-    //
-    /// The path to "/etc/resolv.conf"
-    pub resolv_conf_path: PathBuf,
-
-    /// Optional minimum TTL for DNS lookups.
-    pub dns_min_ttl: Option<Duration>,
-
-    /// Optional maximum TTL for DNS lookups.
-    pub dns_max_ttl: Option<Duration>,
-
-    pub dns_canonicalize_timeout: Duration,
-
-    pub h2_settings: H2Settings,
-}
-
-/// Configuration settings for the tap server
-#[derive(Debug)]
-pub struct TapSettings {
-    pub listener: Listener,
-    pub tap_svc_name: identity::Name,
-}
-
-/// Configuration settings for binding a listener.
-///
-/// TODO: Rename this to be more inline with the actual types.
 #[derive(Clone, Debug)]
-pub struct Listener {
-    /// The address to which the listener should bind.
-    pub addr: SocketAddr,
+pub struct ServerConfig<O: OrigDstAddr = NoOrigDstAddr> {
```

Comment:
I'm _very_ pleased to see that this refactor has finally happened!

It does seem like an enormous diff that could have potentially gone in its own branch, but if that's more work than having it in this PR, don't feel pressured to factor it out...

Reply:
I began to try to split things apart but it's extremely annoying, as it touches changes in transport, which touches changes in io... I can give it another shot, because I'd like to changes to be smaller, but I'm not all that optimistic.